### PR TITLE
feat: maxChar prop for textarea

### PR DESCRIPTION
**Pull request by ramonmulia**
### PR for adding maxChar property in textarea

- add global CSS classes
- add stories
- add tests
- add example in Wagtail, vite, nextjs
- add documentation

new prop:
`maxChar`

story:
Screenshot 2024-10-17 at 09 01 03

### CODE REVIEW

**Comment by BogdanDanilescu** [comment]
edge case scenario, not sure if we should tackle it but just pointing out. If the user decides on having our textarea component and a normal `textarea` html tag we will get an error in the console

Screenshot 2024-10-17 at 18 37 52

again, not sure if it applies to a real scenario because I don't see a reason why the user would use both but just pointing it out. Otherwise, LGTM. Well done @ramonmulia

**Review by BogdanDanilescu — APPROVED** [review]

**Review by hamza14khan — APPROVED** [review]

**Review by pregno — CHANGES_REQUESTED** [review]
@ramonmulia just a minor change after @BogdanDanilescu input

File: packages/html/ds/src/textarea/textarea.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  getAllTextarea: NodeListOf<HTMLTextAreaElement>;
+
+  constructor(options: TextareaProps) {
+    super(options);
+    this.getAllTextarea = document.querySelectorAll('textarea');
+
+    this.calculateRemainingChars = () => {
+      for (const textarea of this.getAllTextarea) {
+        const { id, value, maxLength } = textarea;
+        const remainingCharsContainer = document.querySelector(
```
[the remark this answers is not in the document]
Comment (ramonmulia) [reply packages/html/ds/src/textarea/textarea.ts]: try catch added.

File: packages/html/ds/src/textarea/textarea.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+export class Textarea extends BaseComponent<TextareaProps> {
+  calculateRemainingChars: () => void;
+  getAllTextarea: NodeListOf<HTMLTextAreaElement>;
+
+  constructor(options: TextareaProps) {
+    super(options);
+    this.getAllTextarea = document.querySelectorAll('textarea');
+
+    this.calculateRemainingChars = () => {
+      try {
```
Comment (pregno) [inline packages/html/ds/src/textarea/textarea.ts:18]: Put it inside the for-loop.
In this way, if you have two textarea, the first without the counter and the second with the counter, it will quit too soon

**Review by pregno — CHANGES_REQUESTED** [review]

File: packages/html/ds/src/textarea/textarea.ts (hunk 2 above)
Comment (ramonmulia) [reply packages/html/ds/src/textarea/textarea.ts]: good catch! fixed.

**Review by pregno — APPROVED** [review]
